packages/svelte-materialify/src/components/Textarea/Textarea.svelte

```diff
@@ -30,37 +30,39 @@
   export let style = null;
   export let textarea = null;
 
-  let labelActive = !!placeholder || value;
+  let focused = false;
+  $: labelActive = !!placeholder || value || focused;
   let errorMessages = [];
 
-  function checkRules() {
+  export function validate() {
     errorMessages = rules.map((r) => r(value)).filter((r) => typeof r === 'string');
     if (errorMessages.length) error = true;
     else {
       error = false;
     }
+    return error;
   }
 
   function onFocus() {
-    labelActive = true;
+    focused = true;
   }
 
   function onBlur() {
-    if (!value && !placeholder) labelActive = false;
-    if (validateOnBlur) checkRules();
+    focused = false;
+    if (validateOnBlur) validate();
   }
 
   function clear() {
     value = '';
-    if (!placeholder) labelActive = false;
   }
 
   function onInput() {
-    if (!validateOnBlur) checkRules();
-    if (autogrow) {
-      textarea.style.height = 'auto';
-      textarea.style.height = `${textarea.scrollHeight}px`;
-    }
+    if (!validateOnBlur) validate();
+  }
+  
+  $: if (autogrow) {
+    textarea.style.height = 'auto';
+    textarea.style.height = `${textarea.scrollHeight}px`;
   }
```

Comment:
I was concerned that is the value is dynamically changed to a big value, that the `autogrow` wouldn't get applied, so I made that one dynamic too ; is this the right approach?